apps/wallet-mobile/src/features/AddWallet/common/ButtonCard/ButtonCard.tsx

```diff
@@ -0,0 +1,125 @@
+import {useTheme} from '@yoroi/theme'
+import React from 'react'
+import {StyleSheet, Text, TouchableOpacity, View} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+
+import CreateWallet from '../../illustrations/CreateWallet'
+import HardwareWallet from '../../illustrations/HardwareWallet'
+import RestoreWallet from '../../illustrations/RestoreWallet'
+
+type IconProps = 'create' | 'restore' | 'hardware'
+
+type ButtonCardProps = {
+  title: string
+  subTitle?: string
+  icon?: IconProps
+  onPress: () => void
+}
+
+export const ButtonCard = ({title, subTitle, icon, onPress}: ButtonCardProps) => {
+  const {styles, colors} = useStyles(icon)
+
+  function splitInLines(text: string): string {
+    const words = text.trim().split(' ')
+    let result = ''
+
+    if (words.length <= 2) {
+      result = text
+    } else if (words.length === 3) {
+      result = `${words[0]}\n${words[1]} ${words[2]}`
+    } else {
+      const halfIndex = Math.ceil(words.length / 2)
+      const firstLine = words.slice(0, halfIndex).join(' ')
+      const secondLine = words.slice(halfIndex).join(' ')
+      result = `${firstLine}\n${secondLine}`
+    }
+
+    return result
+  }
+
+  const IconComponent = (component?: IconProps) => {
```

Comment:
it is a static decision it doesn't need to be a function, default should be a view prefer:
```
default:
hardware:
  yadayada
```
if the icon is in fact optional return null